### DISCUSSION TITLE
How do you restoreBranches that You Deleted from the /branches screen in a repo in Github?

### DISCUSSION THREAD

**Discussion by kchalmershappymoney** [community/community#55884] [Repositories] [answered]
### Select Topic Area

Question

### Body

Hello - I deleted a branch from within the "branches" web page in my repo on the Github webpage. The branch was called "add_new_databricks_jobs" - I cannot find documentation for how you can recover a branch specifically when you deleted it this way. Can you help please?

Thank you!

![image](https://github.com/community/community/assets/118197663/156ffa24-f5cc-4cb1-a407-535aab7c5896)

**Comment by samuelss1102** [comment]
When you delete a branch, you are only deleting a reference to the commit at the tip of that branch. The commits themselves are still present in the repository.

Here's a step-by-step guide on how to recover a deleted branch. This assumes that you have at least a clone of the repository with the commits which were in the branch:

Open a terminal and change to the directory where your project is located.

Use the following command to look at all commits across all branches, along with their comments:

`git log --all --decorate --oneline --graph`

This will give you a long list of every commit made in every branch in your repository.

Look for the commit that was the head of the deleted branch. This will be the last commit that was made to the branch. You should recognize it by the commit message. Once you find it, make a note of the commit hash (the long string of numbers and letters next to each commit).

Once you have the hash of the commit that was at the tip of the branch, use the following command to create a new branch with this commit as the head:

`git checkout -b add_new_databricks_jobs <commit-hash>`

Replace <commit-hash> with the actual hash of the commit.

This command will create a new branch with the same name as the deleted one and move you into it.

If you want to push this branch to the remote repository, use the following command:

`git push origin add_new_databricks_jobs`

That's it! Your branch should be restored.

Please note that these instructions will only work if the commits that were on the branch still exist in your local repository. If you have garbage collected your repo (git gc), or if the branch was only present in a different clone of the repository that you do not have access to, then this process will not work. In that case, you would need to find a copy of that branch in a different clone of the repository.

Also, be aware that if you had any uncommitted changes in your working directory or index when the branch was deleted, those changes would not have been part of any commit, and thus will not be recoverable using this method.

**Comment by Bechuana1** [comment]
hello, the **easiest solution** is to Open the repository in GitHub where the branch was deleted.

 1. Click on the "Code" tab to view the repository's code.

 2. In the dropdown menu next to the branch selector, click on the branch selector and select the branch that was deleted. 
   > GitHub will display a message indicating that the branch was deleted and provide an option to **restore** it. 

 3. Click on the "Restore branch" button.
 4. GitHub will recreate the deleted branch with all its commit history.

**Comment by allanjs88** [comment, marked as answer]
Hello,

1. In Github go to Activity
![image](https://github.com/community/community/assets/15911154/2fe1f1cc-a1d9-423a-a5a1-50b9f6efb985)
2. Select Branch Deletions 
![image](https://github.com/community/community/assets/15911154/c87f9ce6-dd49-4453-bf0b-24ff71c07ffa)
3. Look the name of the branch and click in ... at the end 
![image](https://github.com/community/community/assets/15911154/f4826177-9a84-4320-9d07-1b9f8ae349c5)
4. Click in Restore Branch

**Reply by tashrifbillah** [reply]
This is the best answer in this discussion.

**Reply by momothain** [reply]
^

**Reply by g9wz** [reply]
you just saved me two days worth of work :)

**Reply by 0xMaxii** [reply]
You just saved me from a lot of headaches now, thank you so much.

**Reply by MakePixelsWork** [reply]
Thank God (and @allanjs88) for this answer!

**Reply by crawlive** [reply]
life saver!

**Reply by nasomu-fadv** [reply]
after lot of googling found this page, thanks for the info @allanjs88

**Reply by EclipsesDev** [reply]
thx my g

**Reply by iarc13** [reply]
Thanks! Best way to track!

**Reply by OsamaEid1** [reply]
You saved my life!

**Reply by Adrian08796** [reply]
Lifesaver!

**Reply by yashspr** [reply]
Lifesaver. Nothing else worked, no reflog, no log, nothing. This worked! So glad i didn't give up and found this.

**Reply by stephanschoenig** [reply]
Indeed, a lifesaver!  Thank you [allanjs88](https://github.com/allanjs88)

**Reply by csandman** [reply]
This is definitely the best solution provided here, thank you!

**Reply by xharshvardhann** [reply]
lifesaver

**Reply by Jan961** [reply]
Thank you!

**Reply by GBG7** [reply]
ur my new goat. saved my entire project. <3

**Reply by bw2k** [reply]
Thank you man. you saved my day!

**Reply by imdangle** [reply]
Life saver!!!! Thank you!

**Reply by levan563** [reply]
Thank you! Life saver!

**Reply by kr1sjanee** [reply]
tysm, because someone deleted the dev branch; it seems she weren't in her right mind.

**Reply by tatatupi** [reply]
Thanks! I was trying to find an old commit, but it didn't show using `$ git log --all --grep="hash"`

**Reply by OlolaJaco** [reply]
Thank you very much, i am here about to faint. A branch i spent days on and my deadline is in few hours

**Reply by dvijayak** [reply]
Thank you SO much!

**Reply by Near-dev-Pro** [reply]
You saved my day...!!!

**Reply by Vaishnavi20011225** [reply]
Thank you so much!! You have no idea how much we are grateful to you!!

**Reply by ananyaanandh-beckman** [reply]
Thanks! This worked like a charm.

**Reply by 6034903** [reply]
thanks saved my day

**Reply by Rahim-Ktata** [reply]
not all heros wear capes

**Reply by PrinstonZod** [reply]
Thank you so much, man. Just saved my job

**Reply by MichaelKazembe** [reply]
i was stressing over this after my code broke and i didnt know to recover some code. Thank you very much

**Reply by Aspiand** [reply]
Thanks

**Reply by Benjafo** [reply]
Thank you @allanjs88, you're a lifesaver

**Reply by synasapmob** [reply]
that's really cool things dude, in the moment pannic I'm used git push again (in local, of course **tree** is latest) xDDD

**Reply by imnhasan** [reply]
Thanks a lot, save my time

**Reply by Marcono1234** [reply]
**Important:** There seems to be currently a bug when you reuse branch names. In that case GitHub erroneously restores the first version with that name (possibly months old), and not the one you have just deleted.

(Have reported this to GitHub Support already now.)

**Comment by pankaj4u4m** [comment]
Goto all branched and click activity of any one branch
![image](https://github.com/user-attachments/assets/86d5dfac-c6f9-4287-bc72-b6f3278b441e)

Then click activity from all branched 
![image](https://github.com/user-attachments/assets/533280ce-444d-4839-aecf-788b76e10e8b)


then click ... to restore deleted branch

**Reply by Bikram-DoorDash** [reply]
Here, some deleted branches have only "Compare changes" option while others have "restore branch" option.

**Comment by NyxianNetwork** [comment]
To recover a deleted branch on GitHub, you can try the following steps:

1. **Check Local Repository**: If you have a local copy of the branch, you can simply push it back to GitHub:
   ```bash
   git push origin add_new_databricks_jobs
   ```

2. **Use Git Reflog**: If the branch was deleted locally, you can use `git reflog` to find the last commit reference of the branch:
   ```bash
   git reflog
   ```
   Look for the last commit hash associated with "add_new_databricks_jobs," then create a new branch from it:
   ```bash
   git checkout -b add_new_databricks_jobs <commit-hash>
   git push origin add_new_databricks_jobs
   ```

3. **Contact GitHub Support**: If you don’t have a local copy and can't find the commit hash, GitHub Support may be able to help recover the branch history for you.

**Comment by Rahim-Ktata** [comment]
Damn bro , you also saved my project

---------- Forwarded message ---------
From: Eugene Ofarn-Uba ***@***.***>
Date: Thu, 17 Apr 2025 at 17:00
Subject: Re: [community/community] How do you restoreBranches that You
Deleted from the /branches screen in a repo in Github? (Discussion #55884)
To: community/community ***@***.***>
Cc: Ktata Abderrahim ***@***.***>, Comment <
***@***.***>


Thank you so much, man. Just saved my job

—
Reply to this email directly, view it on GitHub
<https://github.com/orgs/community/discussions/55884#discussioncomment-12869426>,
or unsubscribe
<https://github.com/notifications/unsubscribe-auth/BONIOK7ZYJ75T5FTFQQXP3T2Z7FZJAVCNFSM6AAAAAAYHA4CPKVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTEOBWHE2DENQ>
.
You are receiving this because you commented.Message ID:
***@***.***>

**Comment by imeligilani** [comment]
You saved my project man.

**Comment by Nu-man-71** [comment]
go to activities and branch detection and restored
